widgets/bat_openbsd.lua

```diff
@@ -0,0 +1,74 @@
+-- {{{ Grab environment
+local setmetatable = setmetatable
+local tonumber = tonumber
+local string = { format = string.format, gmatch = string.gmatch }
+local io = { popen = io.popen }
+local math = { floor = math.floor }
+local helpers = require("vicious.helpers")
+local string = {
+    gmatch = string.gmatch,
+    match = string.match,
+    format = string.format
+
+}
+-- }}}
+local bat_openbsd = {}
+
+local function worker(format, warg)
+    local battery = warg or "bat0"
+    local filter = string.format("hw.sensors.acpi%s", battery)
+    local cmd = string.format("sysctl -a | grep '^%s'", filter)
+    local proc = io.popen(cmd)
+
+    local bat_info = {}
+    for line in proc:lines("*l") do
+        for key, value in string.gmatch(line, "(%S+)=(%S+)") do
+	    key = key:gsub(filter .. ".", "")
+	    bat_info[key] = value
+        end
+    end
+    proc:close()
```

Comment:
`io.lines` closes the file after the iteration completes.

Reply:
TIL! Well, one less line to maintain.

Reply:
My mistake: while `io.lines(file)` does closes `file` after the iteration, `file:lines()` does not. I will fix this along with the refactoring to completely get rid of `popen`.